src/metabase/api/common/openapi.clj

```diff
@@ -86,21 +101,55 @@
     (->> (walk {:prefix ""} root)
          (filter #(-> % :route meta :schema)))))
 
+(defn- compojure-query-params
+  "This function is not trying to parse whole compojure syntax, just get the names of query parameters out"
+  [args]
+  (loop [args args
+         params []]
+    (if (empty? args)
+      (map keyword params)
+      (let [[x y] args]
+        (cond
+          ;; we don't care about rest binding
+          (= '& x)    (recur (nnext args) params)
+          ;; and about coercion too
+          (= :<< x)   (recur (nnext args) params)
+          (= :as x)   (recur (nnext args)
+                             (into params (when (map? y)
+                                            (let [qp (:query-params (set/map-invert y))]
+                                              ;; {c :count :keys [a b}} ; => [count a b]
+                                              (flatten (vals qp))))))
```

Comment:
Wow this is sneaky haha. This will produce some weird junk if there's an `:or`, I think. I'd prefer if we handled the magic keys separately, and avoided flatten. How about something like:

```clojure
(concat (vals (dissoc qp :or :strs :syms :keys :as))
        (:keys qp))
```

Reply:
Hmm... Let's backport as it is for now maybe?